### PR TITLE
Add tests for material banner example

### PR DESCRIPTION
**Pull request by derdilla**
Part of https://github.com/flutter/flutter/issues/130459.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

### CODE REVIEW

File: examples/api/test/material/banner/material_banner.0_test.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/banner/material_banner.0.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('shows all elements', (WidgetTester tester) async {
```
Comment (bleroux) [inline examples/api/test/material/banner/material_banner.0_test.dart:11]: Consider capitalizing the first letter of test descriptions (same for the other tests below and your other opened PRs).

**Review by bleroux — CHANGES_REQUESTED** [review]
Thanks for adding these valuable tests.
Just one small comment about capitalizing test descriptions first letter.

**Review by bleroux — APPROVED** [review]
LGTM! Thanks for adding those tests.
Taha and I are attending Google/IO this week, expect slow review, we will catch up on reviews next week.

**Comment by bleroux** [comment]
@NobodyForNothing Thanks for your patience, can you rebase this PR and solve the conflict if there is still one?

**Comment by bleroux** [comment]
Thanks for the rebase.
Linux analyze step failed because of several trailing space characters, see the corresponding lines in https://ci.chromium.org/ui/p/flutter/builders/try/Linux%20analyze/79407/overview

File: examples/api/test/material/banner/material_banner.1_test.dart [hunk 2]
```diff
@@ -3,10 +3,36 @@
 // found in the LICENSE file.
 
 import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/material/banner/material_banner.1.dart'
+    as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+
```
Comment (bleroux) [inline examples/api/test/material/banner/material_banner.1_test.dart:11]: Those lines were duplicated, probably during the conflict resolution.

**Review by bleroux — CHANGES_REQUESTED** [review]

**Review by bleroux — APPROVED** [review]
LGTM!

File: examples/api/test/material/banner/material_banner.0_test.dart [hunk 3]
```diff
@@ -7,6 +7,21 @@ import 'package:flutter_api_samples/material/banner/material_banner.0.dart' as e
 import 'package:flutter_test/flutter_test.dart';
 
 void main() {
+  testWidgets('Shows all elements', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.MaterialBannerExampleApp()
```
Comment (TahaTesser) [inline examples/api/test/material/banner/material_banner.0_test.dart:12]:
```suggestion
      const example.MaterialBannerExampleApp(),
```

File: examples/api/test/material/banner/material_banner.0_test.dart [hunk 4]
```diff
@@ -18,4 +33,16 @@ void main() {
     expect(find.widgetWithText(TextButton, 'OPEN'), findsOne);
     expect(find.widgetWithText(TextButton, 'DISMISS'), findsOne);
   });
+
+  testWidgets('The banner is below the text saying so', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const example.MaterialBannerExampleApp()
```
Comment (TahaTesser) [inline examples/api/test/material/banner/material_banner.0_test.dart:39]:
```suggestion
      const example.MaterialBannerExampleApp(),
```

File: examples/api/test/material/banner/material_banner.1_test.dart [hunk 5]
```diff
@@ -3,10 +3,31 @@
...
+    expect(find.byIcon(Icons.agriculture_outlined), findsNothing);
+
+    await tester.tap(find.text('Show MaterialBanner'));
+    await tester.pumpAndSettle();
+    expect(find.byType(MaterialBanner), findsOneWidget);
+    expect(find.text('DISMISS'), findsOneWidget);
+    expect(find.byIcon(Icons.agriculture_outlined), findsOneWidget);
+
+    final MaterialBanner banner = tester.widget<MaterialBanner>(
+        find.byType(MaterialBanner));
```
Comment (TahaTesser) [inline examples/api/test/material/banner/material_banner.1_test.dart:27]:
```suggestion
    final MaterialBanner banner = tester.widget<MaterialBanner>(
      find.byType(MaterialBanner),
    );
```

**Review by TahaTesser — APPROVED** [review]
LGTM! Just minor format issues

**Pull request merged**